tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/SdkRepoConfigHelper.cs

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.Sdk.Tools.Cli.Helpers
+{
+    public interface ISdkRepoConfigHelper
+    {
+        Task<string> GetConfigFilePathForRepoAsync(string repoName);
+        Task<SdkRepoConfiguration> GetRepoConfigurationAsync(string repoName);
+        
+        // Config value retrieval methods
+        Task<T> GetConfigValueFromRepoAsync<T>(string repositoryRoot, string repoName, string jsonPath);
+        Task<(BuildConfigType type, string value)> GetBuildConfigurationAsync(string repositoryRoot, string repoName);
+        
+        // Command processing methods
+        string SubstituteCommandVariables(string command, Dictionary<string, string> variables);
+        string[] ParseCommand(string command);
+    }
+
+    public class SdkRepoConfigHelper : ISdkRepoConfigHelper
+    {
+        // JSON path constants
+        private const string BuildCommandJsonPath = "packageOptions/buildScript/command";
+        private const string BuildScriptPathJsonPath = "packageOptions/buildScript/path";
+
+        private readonly ILogger<SdkRepoConfigHelper> _logger;
+        private readonly string _sdkRepoConfigPath;
+        private Dictionary<string, SdkRepoConfiguration>? _cachedConfig;
+
+        public SdkRepoConfigHelper(ILogger<SdkRepoConfigHelper> logger)
+        {
+            this._logger = logger;
+            this._sdkRepoConfigPath = Path.Combine(
+                AppDomain.CurrentDomain.BaseDirectory, 
+                "Configuration", 
+                "SdkRepoConfig.json");
```

Comment:
What's the path to the configure file? Is it `<repo_root>/Configuration/SdkRepoConfig.json`? If so, should we put it into somewhere within `eng` folder instead?